File: 560-subarray-sum-equals-k/subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        currSum, count = 0,0
        prefixMap = defaultdict(int)
        prefixMap[0] = 1
        for i,val in enumerate(nums):
            currSum += nums[i]
            seenPrefixSum = currSum - k
            if seenPrefixSum in prefixMap:
                count += prefixMap[seenPrefixSum]
            prefixMap[currSum] += 1
        return count

File: 560-subarray-sum-equals-k/subarray-sum-equals-k.go
func subarraySum(nums []int, k int) int {
    prefixSumMap := make(map[int]int)
    prefixSumMap[0] = 1

    currSum := 0
    count :=0
    for _, num := range nums {
        currSum += num
        if freq, ok := prefixSumMap[currSum-k]; ok {
            count += freq
        }
        prefixSumMap[currSum]++
    }
    return count
}